python-libraries/nanover-omni/src/nanover/omni/ase.py

```diff
@@ -217,4 +219,13 @@ def make_regular_frame(self):
         if self.dynamics is not None:
             frame_data.simulation_time = self.dynamics.get_time() * ASE_TIME_UNIT_TO_PS
 
+        # Add user forces and energies to frame
+        # (convert back from ASE internal units to NanoVer units)
+        user_sparse_indices, user_sparse_forces = get_sparse_forces(
+            self.atoms.calc.total_user_forces
+        )
+        frame_data.user_forces_sparse = user_sparse_forces * (EV_TO_KJMOL / ANG_TO_NM)
+        frame_data.user_forces_index = user_sparse_indices
+        frame_data.user_energy = self.atoms.calc.total_user_energy * EV_TO_KJMOL
```

Comment:
I think it probably makes sense to put this in https://github.com/IRL2/nanover-protocol/blob/c2a60435d495875b94a1b6677e7f3ee49051fb7b/python-libraries/nanover-ase/src/nanover/ase/converter.py#L65-L120 since it depends only on `atoms`

Reply:
I think it was the fact that it assumed an iMD calculator that made me put it where it is. Maybe it would be sensible to add an if statement to check that the calculator is an iMD calculator, what do you think?